src/util/userHelpers.js

```diff
@@ -0,0 +1,122 @@
+ * @param {String} targetUserType Check that the extended data is relevant for this user type.
+ * @param {Object} userFieldConfigs an extended data configurtions for user fields.
+ * @returns Array of picked extended data fields
+ */
+export const initialValuesForUserFields = (data, targetScope, targetUserType, userFieldConfigs) => {
+  return userFieldConfigs.reduce((fields, field) => {
+    const { key, userTypeConfig, scope = 'public', schemaType } = field || {};
+    const namespacedKey = getNamespacedKey(scope, key);
+
+    const isKnownSchemaType = EXTENDED_DATA_SCHEMA_TYPES.includes(schemaType);
+    const isTargetScope = scope === targetScope;
+    const isTargetUserType =
+      !userTypeConfig?.limitToUserTypeIds || userTypeConfig?.userTypeIds?.includes(targetUserType);
+
+    if (isKnownSchemaType && isTargetScope && isTargetUserType) {
+      const fieldValue = getFieldValue(data, key);
+      return { ...fields, [namespacedKey]: fieldValue };
+    }
+    return fields;
+  }, {});
+};
+
+/**
+ * Returns input fields for custom user fields
+ * @param {*} props
+ * @returns an array of <CustomExtendedDataField> components
+ */
+export const getCustomUserFieldInputs = (userFieldsConfig, intl, userType = null) => {
+  return userFieldsConfig.reduce((pickedFields, fieldConfig) => {
+    const { key, userTypeConfig, schemaType, scope } = fieldConfig || {};
+    const namespacedKey = getNamespacedKey(scope, key);
+
+    const isKnownSchemaType = EXTENDED_DATA_SCHEMA_TYPES.includes(schemaType);
+    const isTargetUserType =
+      !userTypeConfig?.limitToUserTypeIds || userTypeConfig?.userTypeIds?.includes(userType);
+    const isProviderScope = ['public', 'private', 'protected'].includes(scope);
+
+    return isKnownSchemaType && isTargetUserType && isProviderScope
+      ? [
+          ...pickedFields,
+          {
+            key: namespacedKey,
+            name: namespacedKey,
+            fieldConfig: fieldConfig,
+            defaultRequiredMessage: intl.formatMessage({
+              id: 'CustomExtendedDataField.required',
+            }),
+          },
+        ]
+      : pickedFields;
+  }, []);
+};
+
+/**
+ * Get the namespaced attribute key based on the specified extended data scope and attribute key
+ * @param {*} scope extended data scope
+ * @param {*} key attribute key in extended data
+ * @returns a string containing the namespace prefix and the attribute name
+ */
+const getNamespacedKey = (scope, key) => {
+  const namespacePrefix = scope === 'public' ? `pub_` : scope === 'protected' ? 'prot_' : `priv_`;
+
+  return `${namespacePrefix}${key}`;
+};
+
+/**
+ * Returns the value of the attribute in extended data.
+ * @param {*} data extended data containing the value
+ * @param {*} key attribute key in extended data
+ * @returns
+ */
+const getFieldValue = (data, key) => {
+  return data && data[key] ? data[key] : data && typeof data[key] === 'boolean' ? false : null;
```

Comment:
the original code would return _null_, if 
- data[key] resolves to 0, ([falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value in JS)
- data[key] resolves to "", (also [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy))

Reply:
The use case for this is displaying a value that is explicitly false, not null or undefined – i.e. "canCook" has a value "false" in publicData (and it gets saved correctly), but when the value is retrieved with `fieldValue = publicData[key] || null`, an existing boolean 'false' does not get resolved correctly.

I noticed that this helper is actually necessary in ProfilePage as well, when displaying a boolean false value, so I moved it from userHelpers to fieldHelpers, but I can definitely update it to e.g.

`return data?.[key] ? data[key] : typeof data?.[key] === 'boolean' ? false : null;`

Reply:
@SariSaar If it's for displaying content, then you should change it so that only null or undefined are cleared away:

```
  const value = data?.[key];
  return value != null ? value : null;
```

Having this structure: `data?.[key] ? data[key] : null`, would return null for any falsy value.

Reply:
For publicData.canCook: false, `data?.[key] ? data[key] : null` returns `null`, even though we want to return `false`, because the ternary discards all falsy values. So the logic here would be to explicitly check for a falsy boolean, because that's a valid value to display.

.